### ISSUE TITLE
Build on Summit

### ISSUE THREAD

**Issue by LDAmorim** [ECP-WarpX/WarpX#1453]
Hi everyone,

Sorry if we discussed this issue recenly and I forgot how its fixed (or how to correctly compile the current code versions in Summit).

I tried to compile WarpX in Summit@OLCF and got the following warning message as the end of the output:
```
./Source/Parallelization/WarpXRegrid.cpp: In member function 'void WarpX::LoadBalance()':
./Source/Parallelization/WarpXRegrid.cpp:80:19: warning: conversion to 'std::vector<int, std::allocator<int> >::size_type {aka long unsigned int}' from 'amrex::Real {aka double}' may alter its value [-Wfloat-conversion]
                 pmap.resize(nboxes);
```

To double check this issue I deleted my codes versions and re-cloned them from github and then I tried to compile WarpX with the commands below:
```
module load cuda
module load gcc/6.4.0
make realclean USE_GPU=TRUE
make -j 16 COMP=gcc USE_GPU=TRUE DIM=2
```

Do any of you know this issue and how to fix it?

ps: why is `cmake` a required package in the Docs when the suggested compilation command is with `make`?

Thanks,
Diana

**Comment by ax3l** [comment]
Hi Diana,

the warning can be ignored but should be fixed.
https://github.com/ECP-WarpX/WarpX/blob/bc3bf28155c3257e7fd6386bd43a9f9fd52b7287/Source/Parallelization/WarpXRegrid.cpp#L80
If you like you can take this on: `nboxes` is stored as a `amrex::Real` because it is used above in a floating point operation. If you want to fix this in-place in line 80, you could add a `static_cast<std::size_t>()` around it and include `<cstddef>` alongside `<memory>` at the beginning of the file.
```diff
diff --git a/Source/Parallelization/WarpXRegrid.cpp b/Source/Parallelization/WarpXRegrid.cpp
index 1fdfd6564..34bf5cfde 100644
--- a/Source/Parallelization/WarpXRegrid.cpp
+++ b/Source/Parallelization/WarpXRegrid.cpp
@@ -77,7 +77,7 @@ WarpX::LoadBalance ()
                 pmap = newdm.ProcessorMap();
             } else
             {
-                pmap.resize(nboxes);
+                pmap.resize(static_cast<std::size_t>(nboxes));
             }
             ParallelDescriptor::Bcast(&pmap[0], pmap.size(), ParallelDescriptor::IOProcessorNumber());
```

The documentation for Summit shows a complete list of modules with which you can build with GNUmake and with CMake. Due to that, it just loads CMake pro-actively. We generally will transition further to CMake, I recommend to build with CMake on Summit already, this will also ease in the use of newer features (e.g. simple enabling of openPMD with a single switch and Ascent support when needed). Indeed, we should replace the example
```bash
make -j 16 COMP=gcc USE_GPU=TRUE USE_OPENPMD=TRUE
```
in that page. Do you want to update this together with the build link below it (point to CMake page)? :)
```
mkdir -p build
cd build
cmake .. -DWarpX_OPENPMD=ON -DWarpX_DIMS=3 -DWarpX_COMPUTE=CUDA -DCMAKE_CUDA_ARCHITECTURES=70 -DCUDA_ARCH=7.0
make -j 10
```

`-DCMAKE_CUDA_ARCHITECTURES=70 -DCUDA_ARCH=7.0`: Just set because there is currently a tiny verbose warning on Summit with CMake 3.18 that I need to address (but it does no harm): #1459

**Comment by LDAmorim** [comment]
Hi @ax3l,
Thank you so much for the detailed reply and fix!

I will be very happy to apply both the warning fix and the update to the documentation (and I will link the PRs to this issue).

And thank you @mrowan137 for following up on the related question in `slack`.

Cheers,
Diana

**Comment by ax3l** [comment]
Fixed with #1461 and #1462, thank you @LDAmorim ! :tada:
